File: P169 Majority Element/Solution.py
class Solution(object):
    def majorityElement(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 1:
            return nums[0]
        nums.sort()
        majority = nums[0]
        count = 1
        for i in range(1,len(nums)):
            if (nums[i] == majority):
                count += 1
                if(count > len(nums)/2):
                    return nums[i]
            else:
                majority = nums[i]
                count = 1
        return 0

File: P169 Majority Element/Solution.java
public class Solution {
    public int majorityElement(int[] nums) {
        if (nums.length==1)
            return nums[0];
        Arrays.sort(nums);
        int majority = nums[0];
        int count=1;
        for(int i=1; i<nums.length; i++){
            if (nums[i] == majority) {
                count++;
                if(count > nums.length/2) return nums[i];
            }
            else {
                majority = nums[i];
                count = 1;
            }
        }
        return 0;
    }
}